query/control/controller_test.go

```diff
@@ -40,8 +40,16 @@ func init() {
 var (
 	mockCompiler = &mock.Compiler{
 		CompileFn: func(ctx context.Context) (flux.Program, error) {
+			prev := time.Now()
+			for now := time.Now(); now.Equal(prev); now = time.Now() {
```

Comment:
What does this do since the `stats.QueueDuration == 0` checks are being removed? Wondering if it's still needed, and if so if it is checking anything significant.

Reply:
This is making sure `stats.CompileDuration` is nonzero (and the one below is making sure `stats.ExecuteDuration` is nonzero)

Reply:
Gotcha...I won't stress too much about how significant those checks may or may not be, since this change seems to accomplish its goal without too much disruption to stuff that is already working fine.